### PR TITLE
Get authors of secret groups 

### PR DESCRIPTION
**Pull request by MartinBenediktBusch**
This PR implements the functionality to retrieve the authors and co-authors of a secret group for a given question option. This information will be included in the comments page and or vote page.

The output looks like this:

```
{
      optionId: '842bc51d-9d79-4b56-9ed0-6f396a43281b',
      registrationId: '6aa5082b-090a-48ec-b4e0-61edce72e727',
      groupId: '0cdc5cec-d6ef-4636-8c7c-3f9a6d92654d',
      usersInGroup: [
        {
          id: 'dabab499-af68-4ad9-964e-41536f62fb73',
          username: 'Irina.Huber24',
          firstName: null,
          lastName: null
        },
        {
          id: '035a9d59-e517-409f-a72c-8d771cf22ccf',
          username: 'Pilar.Saeli',
          firstName: null,
          lastName: null
        }
      ]
    }
```

### CODE REVIEW

**Review by diegoalzate — CHANGES_REQUESTED** [review]
small change in response structure and changing the wording from authors -> users as it can be more coherent with future updates

File: src/routers/options.ts [hunk 1]
```diff
@@ -1,14 +1,18 @@
...
 import { isLoggedIn } from '../middleware/is-logged-in';
-import { getOptionCommentsHandler, getOptionHandler } from '../handlers/options';
 
 const router = express.Router();
 
 export function optionsRouter({ dbPool }: { dbPool: PostgresJsDatabase<typeof db> }) {
   router.get('/:optionId', isLoggedIn(dbPool), getOptionHandler(dbPool));
   router.get('/:optionId/comments', isLoggedIn(dbPool), getOptionCommentsHandler(dbPool));
-
+  router.get('/:optionId/authors', isLoggedIn(dbPool), getOptionAuthorsHandler(dbPool));
```
[the remark this answers is not in the document]
Comment (MartinBenediktBusch) [reply src/routers/options.ts:16]: makes sense.

File: src/services/comments.ts [hunk 2]
```diff
@@ -129,3 +129,93 @@ export async function userCanComment(
...
   return true;
 }
+
+type AuthorData = {
+  optionId: string;
+  registrationId: string;
+  userId: string;
+  groupId: string;
+  usersInGroup: string[];
+};
```
[the remark this answers is not in the document]
Comment (MartinBenediktBusch) [reply src/services/comments.ts:139]: Updated

File: src/services/comments.ts [hunk 3]
```diff
@@ -129,3 +129,93 @@ export async function userCanComment(
...
+
+/**
+ * Executes a query to retrieve author data related to a forum question from the database.
+ *
+ * @param {string} optionId - The ID of the question option for which author data is to be retrieved.
+ * @param {PostgresJsDatabase<typeof db>} dbPool - The PostgreSQL database pool instance.
+ * @returns {Promise<AuthorData | null>} - A promise resolving to author data related to the question question or null if no data found.
+ */
+export async function getOptionAuthors(
```
[the remark this answers is not in the document]
Comment (MartinBenediktBusch) [reply src/services/comments.ts:148]: updated

**Comment by MartinBenediktBusch** [comment]
updated response object: 

```
{
      "optionId": "d09a570f-5996-466b-be99-fa8090b7a87f",
      "registrationId": "6d7dfb1e-06f0-4997-8f63-0c62090d4ee6",
      "userId": "a22d3841-646b-4872-b9b6-2561974b70fc",
      "user": {
        "id": "a22d3841-646b-4872-b9b6-2561974b70fc",
        "username": "Santosh_Scholz34",
        "firstName": "Hauwa",
        "lastName": "Cai"
      },
      "groupId": "8e12d0bd-c68f-4bc8-bcbc-fede8430fd6e",
      "group": {
        "id": "8e12d0bd-c68f-4bc8-bcbc-fede8430fd6e",
        "users": [
          {
            "id": "a22d3841-646b-4872-b9b6-2561974b70fc",
            "username": "Santosh_Scholz34",
            "firstName": "Hauwa",
            "lastName": "Cai"
          },
          {
            "id": "86082d4a-8773-4665-8708-254258f3046a",
            "username": "Janet_Nakano89",
            "firstName": "Yoshimi",
            "lastName": "Morozov"
          }
        ]
      }
    }
```

File: src/handlers/options.ts [hunk 4]
```diff
@@ -2,7 +2,7 @@ import { eq } from 'drizzle-orm';
 import { PostgresJsDatabase } from 'drizzle-orm/postgres-js';
 import type { Request, Response } from 'express';
 import * as db from '../db';
-import { getOptionComments } from '../services/comments';
+import { getOptionAuthors, getOptionComments } from '../services/comments';
```
Comment (diegoalzate) [inline src/handlers/options.ts:5]: this import should fail because the function is called: getOptionUsers

File: src/handlers/options.ts [hunk 5]
```diff
@@ -39,3 +39,34 @@ export function getOptionCommentsHandler(dbPool: PostgresJsDatabase<typeof db>)
...
+/**
+ * Retrieves author and co-author data for a given question option created as a secret group.
+ *
+ * @param {PostgresJsDatabase<typeof db>} dbPool - The PostgreSQL database pool instance.
+ * @returns {Function} - An Express middleware function handling the request to retrieve result statistics.
+ * @param {Request} req - The Express request object.
+ * @param {Response} res - The Express response object.
+ * @returns {Promise<Response>} - A promise that resolves with the Express response containing the author data.
+ */
+export function getOptionAuthorsHandler(dbPool: PostgresJsDatabase<typeof db>) {
```
Comment (diegoalzate) [inline src/handlers/options.ts:52]: rename to getOptionUsersHandler

**Review by diegoalzate — APPROVED** [review]

File: src/handlers/options.ts (hunk 5 above)
Comment (MartinBenediktBusch) [reply src/handlers/options.ts]: resolved

**Pull request merged**